components/Modal/index.d.ts

```diff
@@ -27,14 +24,14 @@ export interface ModalProps {
   };
 }
 
-export default class Modal extends Component<ModalProps> {
-  static Header: ModalHeader;
-
-  static Title: Title;
-
-  static HeaderText: HeaderText;
+interface CompoundedModal extends React.FunctionComponent<ModalProps> {
```

Comment:
```suggestion
interface Modal extends React.Component<ModalProps> {
```
Modal is not a functional component

Reply:
React.Component doesn't work with subcomponents, Because it is not possible to type in internal components.

Reply:
Why? Can you show an attachment with the problem?

Reply:
![image-2023-01-27-16-40-39-830](https://user-images.githubusercontent.com/37362664/223747489-b3eeead6-28a5-4fea-82ca-1afeb8f339e0.png)